### PR TITLE
fix interact_sky: cases tpf has no PM, tpf has no coordinate, from tpf.cutout()

### LINKED ISSUE #1086: interact_sky failed with exception when tpf misses some information to display target position
#### Problem description
`interact_sky()` failed with `TypeError: unsupported operand type(s) for *: 'NoneType' and 'CompositeUnit'` when some information is missing in the TargetPixelFile to render the target's position (the black cross in the diagram), e.g., `PMRA` header.

#### Example

```python
import lightkurve as lk

# TPF is TIC 229689154, sector 26
# the TPF does not have PMRA / PMDEC in its header
tpf.interact_sky();
```

The resulting error:
```python
ERROR:tornado.application:Uncaught exception GET /autoload.js?bokeh-autoload-element=3128&bokeh-absolute-url=http://localhost:52565&resources=none (::1)
HTTPServerRequest(protocol='http', host='localhost:52565', method='GET', uri='/autoload.js?bokeh-autoload-element=3128&bokeh-absolute-url=http://localhost:52565&resources=none', version='HTTP/1.1', remote_ip='::1')
Traceback (most recent call last):
  File "c:\pkg\_winnonportables\anaconda3\envs\lkv2_dev\lib\site-packages\tornado\web.py", line 1704, in _execute
    result = await result
  File "c:\pkg\_winnonportables\anaconda3\envs\lkv2_dev\lib\site-packages\bokeh\server\views\autoload_js_handler.py", line 60, in get
    session = await self.get_session()
  File "c:\pkg\_winnonportables\anaconda3\envs\lkv2_dev\lib\site-packages\bokeh\server\views\session_handler.py", line 120, in get_session
    session = await self.application_context.create_session_if_needed(session_id, self.request, token)
  File "c:\pkg\_winnonportables\anaconda3\envs\lkv2_dev\lib\site-packages\bokeh\server\contexts.py", line 218, in create_session_if_needed
    self._application.initialize_document(doc)
  File "c:\pkg\_winnonportables\anaconda3\envs\lkv2_dev\lib\site-packages\bokeh\application\application.py", line 171, in initialize_document
    h.modify_document(doc)
  File "c:\pkg\_winnonportables\anaconda3\envs\lkv2_dev\lib\site-packages\bokeh\application\handlers\function.py", line 132, in modify_document
    self._func(doc)
  File "C:\dev\lightkurve\src\lightkurve\interact.py", line 1240, in create_interact_ui
    tpf, fig_tpf, magnitude_limit=magnitude_limit
  File "C:\dev\lightkurve\src\lightkurve\interact.py", line 497, in add_gaia_figure_elements
    target_ra, target_dec, pm_corrected = _get_corrected_coordinate(tpf)
  File "C:\dev\lightkurve\src\lightkurve\interact.py", line 135, in _get_corrected_coordinate
    pm_ra * pm_unit, pm_dec * pm_unit,
TypeError: unsupported operand type(s) for *: 'NoneType' and 'CompositeUnit'
ERROR:tornado.access:500 GET /autoload.js?bokeh-autoload-element=3128&bokeh-absolute-url=http://localhost:52565&resources=none (::1) 2681.92ms
```

#### Environment

-  platform (e.g. Linux, OSX, Windows): Windows
-  lightkurve version (e.g. 1.0b6): 2.0.10
-  installation method (e.g. pip, conda, source): conda

### PR DESCRIPTION
**Pull request by orionlee**
Closes #1086, #1087, #1089

### CODE REVIEW

**Comment by orionlee** [comment]
The fix for #1087 (unhelpful error when `tpf` has no coordinate) is still a bit hard to process, as it is buried under bokeh's generic exception. Users can find the error if they go through the traceback.

```python
ERROR:tornado.application:Uncaught exception GET /autoload.js?bokeh-autoload-element=1501&bokeh-absolute-url=http://localhost:56371&resources=none (::1)
HTTPServerRequest(protocol='http', host='localhost:56371', method='GET', uri='/autoload.js?bokeh-autoload-element=1501&bokeh-absolute-url=http://localhost:56371&resources=none', version='HTTP/1.1', remote_ip='::1')
Traceback (most recent call last):
  File "C:\dev\lightkurve\src\lightkurve\interact.py", line 392, in add_gaia_figure_elements
    c1 = SkyCoord(tpf.ra, tpf.dec, frame="icrs", unit="deg")
  File "c:\pkg\_winnonportables\anaconda3\envs\lkv2_dev\lib\site-packages\astropy\coordinates\sky_coordinate.py", line 316, in __init__
    frame_cls(**frame_kwargs), args, kwargs)
  File "c:\pkg\_winnonportables\anaconda3\envs\lkv2_dev\lib\site-packages\astropy\coordinates\sky_coordinate_parsers.py", line 301, in _parse_coordinate_data
    _components[frame_attr_name] = attr_class(arg, unit=unit)
  File "c:\pkg\_winnonportables\anaconda3\envs\lkv2_dev\lib\site-packages\astropy\coordinates\angles.py", line 642, in __new__
    self = super().__new__(cls, angle, unit=unit, **kwargs)
  File "c:\pkg\_winnonportables\anaconda3\envs\lkv2_dev\lib\site-packages\astropy\coordinates\angles.py", line 137, in __new__
    return super().__new__(cls, angle, unit, dtype=dtype, copy=copy)
  File "c:\pkg\_winnonportables\anaconda3\envs\lkv2_dev\lib\site-packages\astropy\units\quantity.py", line 383, in __new__
    raise TypeError("The value must be a valid Python or "
TypeError: The value must be a valid Python or Numpy numeric type.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "c:\pkg\_winnonportables\anaconda3\envs\lkv2_dev\lib\site-packages\tornado\web.py", line 1704, in _execute
    result = await result
  File "c:\pkg\_winnonportables\anaconda3\envs\lkv2_dev\lib\site-packages\bokeh\server\views\autoload_js_handler.py", line 60, in get
    session = await self.get_session()
  File "c:\pkg\_winnonportables\anaconda3\envs\lkv2_dev\lib\site-packages\bokeh\server\views\session_handler.py", line 120, in get_session
    session = await self.application_context.create_session_if_needed(session_id, self.request, token)
  File "c:\pkg\_winnonportables\anaconda3\envs\lkv2_dev\lib\site-packages\bokeh\server\contexts.py", line 218, in create_session_if_needed
    self._application.initialize_document(doc)
  File "c:\pkg\_winnonportables\anaconda3\envs\lkv2_dev\lib\site-packages\bokeh\application\application.py", line 171, in initialize_document
    h.modify_document(doc)
  File "c:\pkg\_winnonportables\anaconda3\envs\lkv2_dev\lib\site-packages\bokeh\application\handlers\function.py", line 132, in modify_document
    self._func(doc)
  File "C:\dev\lightkurve\src\lightkurve\interact.py", line 1142, in create_interact_ui
    tpf, fig_tpf, magnitude_limit=magnitude_limit
  File "C:\dev\lightkurve\src\lightkurve\interact.py", line 396, in add_gaia_figure_elements
    raise LightkurveError(msg) from err
lightkurve.utils.LightkurveError: Cannot get nearby stars in GAIA because TargetPixelFile has no valid coordinate. ra: None, dec: None
ERROR:tornado.access:500 GET /autoload.js?bokeh-autoload-element=1501&bokeh-absolute-url=http://localhost:56371&resources=none (::1) 12.96ms
```

Conceivably we could improve it if we catch the implementation so that it throws an exception early, before it calls bokeh. The tradeoff is we would need to add special code paths just for such unlikely cases (TPF has no coordinate at all).

File: src/lightkurve/interact.py [hunk 1]
```diff
@@ -354,6 +354,8 @@ def _add_nearby_tics_if_tess(tpf, source, tooltips):
     if tic_id is None or tic_id == "":
         return source, tooltips
 
+    tic_id = tic_id.replace("_CUTOUT", "")  # for cases tpf is from tpf.cutout() call
```
Comment (orionlee) [inline src/lightkurve/interact.py:357]: The fix for tpf.cutout() is a bit hacky (by removing `_CUTOUT` suffix from the header value). 

I presume there is some reason for `_CUTOUT` to be appended to `TICID` header in the first place, and opt for compensating for the special case here.

File: src/lightkurve/interact.py [hunk 2]
```diff
@@ -350,6 +354,10 @@ def _add_nearby_tics_if_tess(tpf, source, tooltips):
     if tic_id is None or tic_id == "":
         return source, tooltips
 
+    if isinstance(tic_id, str):
+        # for cases tpf is from tpf.cutout() call in #1089
+        tic_id = tic_id.replace("_CUTOUT", "")
```
Comment (orionlee) [inline src/lightkurve/interact.py:359]: The fix for tpf.cutout() is a bit hacky (by removing _CUTOUT suffix from the header value).

I presume there is some reason for _CUTOUT to be appended to TICID header in the first place, and opt for compensating for the special case here.

**Review by barentsen — CHANGES_REQUESTED** [review]
Thank you for these improvements @orionlee!!

My only comment is that I think we ought to modify `TargetPixelFile.cutout()` to stop it from adding the `_CUTOUT` suffix to the `TICID` meta data keyword in the first place.  There is no justification for this suffix that I'm aware of.  Users should reasonably be able to expect that the `TICID` field contains a valid identifier, as opposed to an identifier with an arbitrary string suffix.

I'm happy to push that extra change to this PR if you agree!

**Comment by orionlee** [comment]
@barentsen Feel free to push the extra change to the PR to modify `TargetPixelFile.cutout()` to stop it from adding the `_CUTOUT` suffix .

If we do so, we might want to consider adding another keyword to tpf header signifying it is a result of `cutout()` call. I am of two minds about it though.
- Reason: I guess `_CUTOUT` was added originally because the author wanted to signify that the tpf is a cutout rather than a complete one.
- Counter argument: a tpf file could come from SPOC standard pipeline or TESS Cut service. Currently there is no way to explicitly differentiate them anyway. So adding a keyword to tpf header signifying it is a result of a `cutout` call might be over-engineering.

**Comment by barentsen** [comment]
> Reason: I guess _CUTOUT was added originally because the author wanted to signify that the tpf is a cutout rather than a complete one.

Yes, I think it was done to have so that subsequent calls to `tpf.plot()` would show the word "cutout" in the label of the plot.  This doesn't seem like a good justification, because we are not using target identifier suffixes anywhere else to record operations that have been applied to the data (and indeed files obtained via TessCut do not show this suffix).

I am not aware of any code that depends on the "_CUTOUT" suffix, so I think it is safe to remove.

It could of course be useful to keep track of the data operations applied.  For example, we have been dreaming about populating a `lc.meta["HISTORY"]` keyword with information about the operations applied. That seems like a much better place to record that a cutout has been made.

I pushed the extra change and intend to merge if the tests pass.  Thanks again for fixing this issue!!

**Pull request merged**